### PR TITLE
Recipe App 🥘🍲🥗🍝

### PR DESCRIPTION
**Pull request by joyapisi**
## Work Done:
- Create Kanban board and distribute work between ourselves
- Follow the layout of the wireframes provided.
- Use Devise gem to set up login page and registration page
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.
- Write unit and integration tests

- Implement food list:
  - Display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
  - Set up form that allows users to add new food.
- Recipes list:
  - Display a list of recipes created by the logged-in user as in the wireframe.
  - Leads to recipe details.
  - If the user is the owner of the recipe, it allows the user to delete it.
- Public recipe list:
  - Display a list of all public recipes ordered by the newest as in the wireframe.
  - Can lead to recipe details.
- Recipe details:
  - Should display a toggle button that allows for a recipe to be made public or private.
  - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
  - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view (only for a group that has less than 3 members):
  - Show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
  - Should count the total food items and total price of the missing food.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  # gem "spring"
+end
+
+group :test do
+  # Use system testing [https://guides.rubyonrails.org/testing.html#system-testing]
+  gem 'capybara'
+  gem 'selenium-webdriver'
+  gem 'webdrivers'
+end
+gem 'cancancan'
```
Comment (laiifuu) [inline Gemfile:76]: When I was using the application, there were some resources i was not supposed to access but I could access them: 
- I was able to access the foods list of another user, 
- I was able to delete foods created by another user, 
- I was able to access the list of private recipes of another user, 
- I was able to delete a private recipe created by another user. 
- I was able to make a public recipe created by another user, private, 
You should fix this behavior.

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    t.datetime "updated_at", null: false
+    t.index ["user_id"], name: "index_foods_on_user_id"
+  end
+
+  create_table "recipe_foods", force: :cascade do |t|
+    t.decimal "quantity"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "food_id", null: false
+    t.bigint "recipes_id"
```
Comment (laiifuu) [inline db/schema.rb:33]: This attribute is not necessary, you should remove it so your schema can match the entity relationship diagram provided.

File: app/views/public_recipes/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <br/>
+    <p>
+      By: <%= recipe.user.name %>
+    </p>
+    </div>
+    <% end %>
+    </div>
+
+  <% end %>
+</div>
```
Comment (laiifuu) [inline app/views/public_recipes/index.html.erb:25]: As mentioned in the requirements, the views should match the wireframes provided. For the public recipes page, each publick recipe should show the total number of food items, and the total price for the recipe. 
- The wireframe: 
image
- Your design: 
image

**Review by laiifuu — CHANGES_REQUESTED** [review]
## Hi team :wave:  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

cat-driving-serious

## Highlights
- [x] Linters check is passing :heavy_check_mark: 
- [x] Necessary gems are present in the `GemFile` :heavy_check_mark: 
- [x] `Gitflow` is respected :heavy_check_mark: 
- [x] User registration and login via devise :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job following the instructions of the previous CR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**